perf(transformer/module_runner): use `itoa` to convert `u32` to string

overlookmotel opened the pull request:
Follow-on after #9401. Use `itoa` to convert `u32` to string. It's more optimized for this task than `CompactString`.

overlookmotel commented:
* **#9687** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9687?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9686** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9686?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9686?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9686).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing reviewed (approved):
Looks great, BTW we don't have a benchmark for `ModuleRunnerTransform` as it isn't included in the main transformer

overlookmotel committed:
perf(transformer/module_runner): use `itoa` to convert `u32` to string (#9686)

Follow-on after #9401. Use `itoa` to convert `u32` to string. It's more optimized for this task than `CompactString`.
